server/blob/blob-api/src/main/java/org/apache/james/blob/api/ObjectStorageHealthCheck.java

```diff
@@ -0,0 +1,59 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.blob.api;
+
+import java.time.Duration;
+
+import javax.inject.Inject;
+
+import org.apache.james.core.healthcheck.ComponentName;
+import org.apache.james.core.healthcheck.HealthCheck;
+import org.apache.james.core.healthcheck.Result;
+
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+public class ObjectStorageHealthCheck implements HealthCheck {
+
+    private static final Integer HEALTH_CHECK_TIMEOUT = 10;
+
+    private static final ComponentName COMPONENT_NAME = new ComponentName("ObjectStorage");
+
+    private final BlobStoreDAO blobStoreDAO;
+
+    @Inject
+    public ObjectStorageHealthCheck(BlobStoreDAO blobStoreDAO) {
+        this.blobStoreDAO = blobStoreDAO;
+    }
+
+    @Override
+    public ComponentName componentName() {
+        return COMPONENT_NAME;
+    }
+
+    @Override
+    public Mono<Result> check() {
+        return Flux.from(blobStoreDAO.listBuckets())
```

Comment:
Q: Is it ok if using `listBuckets` for health check in prod?

I don't find an api for healthcheck in aws s3 sdk. 
For Minio supports API for health checks, but it is not generic for all s3 platforms. https://min.io/docs/minio/linux/operations/monitoring/healthcheck-probe.html

Reply:
listBuckets is a relatively cheap request, think it is ok to use it for a healthcheck